src/core/or/circuitmux.c

```diff
@@ -1011,23 +1011,17 @@ circuitmux_clear_num_cells(circuitmux_t *cmux, circuit_t *circ)
 }
 
 /**
- * Set the cell counter for a circuit on a circuitmux
+ * Set the cell counter for a circuit on a circuitmux with a given hashmap
  */
 
 void
-circuitmux_set_num_cells(circuitmux_t *cmux, circuit_t *circ,
-                         unsigned int n_cells)
+circuitmux_set_num_cells_hashent(circuitmux_t *cmux, circuit_t *circ,
+                                 chanid_circid_muxinfo_t *hashent,
+                                 unsigned int n_cells)
 {
-  chanid_circid_muxinfo_t *hashent = NULL;
-
   tor_assert(cmux);
   tor_assert(circ);
 
-  /* Search for this circuit's entry */
-  hashent = circuitmux_find_map_entry(cmux, circ);
-  /* Assert that we found one */
-  tor_assert(hashent);
```

Comment:
There's no reason to delete this assertion; and in fact we might want to have an assertion that the hashent matches cmux and circ.